### PR TITLE
Sync ShaderDerivateTests with upstream CTS

### PR DESCRIPTION
**Pull request by lexaknyazev**
Ported:
- https://github.com/KhronosGroup/VK-GL-CTS/commit/d9551dc8ecff19b0bfe50e0a463c49e99997b4a2
- https://github.com/KhronosGroup/VK-GL-CTS/commit/59b032bc7f837293068a2ee0469274c6dfede801
- https://github.com/KhronosGroup/VK-GL-CTS/commit/1dfe5d68bd78ebc44ee80222452e4a5f5656aab6
- https://github.com/KhronosGroup/VK-GL-CTS/commit/964431bd0878c57dde43978cfda34401b334c409
- https://github.com/KhronosGroup/VK-GL-CTS/commit/ce02996d700551a7fcc2889a55c54b20466b8cae
- https://github.com/KhronosGroup/VK-GL-CTS/commit/07b505e709ff516b426556e4593443b8f496c8c0

Confirmed that the updated tests pass on Adreno.

### CODE REVIEW

**Review by kdashg — APPROVED** [review]

File: sdk/tests/deqp/functional/gles3/es3fShaderDerivateTests.js [hunk 1]
```diff
@@ -284,6 +284,22 @@ goog.scope(function() {
         return value + numUlps * es3fShaderDerivateTests.getSingleULPForValue(value, numMantissaBits);
     };
 
+    /**
+     * @param {?gluShaderUtil.precision} precision
+     * @return {number}
+     */
+    es3fShaderDerivateTests.getInterpolationLostBitsWarning = function(precision) {
+        // number mantissa of bits allowed to be lost in varying interpolation
```
Comment (kenrussell) [inline sdk/tests/deqp/functional/gles3/es3fShaderDerivateTests.js:292]: Maybe an upstream typo? "number of mantissa bits"?

**Review by kenrussell — APPROVED** [review]
Great work porting these tricky test updates from the native suite. Looks good; thank you.

**Comment by kenrussell** [comment]
Merging now; the comment about a comment above isn't important.

File: sdk/tests/deqp/functional/gles3/es3fShaderDerivateTests.js (hunk 1 above)
Comment (lexaknyazev) [reply sdk/tests/deqp/functional/gles3/es3fShaderDerivateTests.js:292]: Yes, this is an upstream typo.

**Pull request merged**